stages/stages.cc

```diff
@@ -145,6 +145,149 @@ void Process(IOBuffer::Block* block, size_t size) {
+  if (activeChannelSwitchTime > 0) --activeChannelSwitchTime;
+
+  // Initial slider positions (set when switching channels). This allows us to determine once the
+  // user has moved a slider sufficiently.
+  // We don't use slider locking, since that feature does not support locking to an arbitrary value.
+  static float initial_slider_positions[] = {0, 0, 0, 0, 0, 0};
+  static float slider_move_threshold = 0.05f;
+
+  // For each envelope feature, whether we are currently using slider values
+  // Slider positions are ignored for the active envelope until the user moves them sufficiently to activate them.
+  // Proactively set them as enabled on bootup, however.
+  static bool slider_enabled[] = {true, true, true, true, true, true};
+
+  // Handle for channel switch presses
+  for (size_t ch = 0; ch < kNumChannels; ch++) {
+    // Check if button is pressed to switch channels or activate all sliders.
+    // Ignore switch presses if recently switched.
+    if (!activeChannelSwitchTime && ui.switches().pressed(ch)) {
+      // Once a switch is pressed, delay processing switches for 1/4 second.
+      // This is essentially a super basic debounce.
+      activeChannelSwitchTime = 1000;
+
+      if (ch == active_envelope) {      
+        // Pressing the active channel enables all sliders and pots
+        fill(&slider_enabled[0], &slider_enabled[size], true);
+      } else {
+        // Pressing an inactive channel switches to that channel
+        // Record initial slider positions after switch and set all sliders to inactive
+        active_envelope = ch;
+        fill(&slider_enabled[0], &slider_enabled[size], false);
+        for (size_t slider_index = 0; slider_index < kNumChannels; ++slider_index) {
+          initial_slider_positions[slider_index] = block->slider[slider_index];
+        }
+      }
+    }
+  }
+
+  // Update envelope parameters for active envelope.
+  //
+  // n.b. cv inputs only apply to active envelope and only if overriding dahdsr.
+  // This allows cv to directly offset the slider position. Otherwise we would need to track separate
+  // "virtual" slider positions for each envelope.
+  if (slider_enabled[0]) {
+    eg[active_envelope].SetDelayLength(block->cv_slider[0] + block->cv[0]);
+  }
+  if (slider_enabled[1]) {
+    eg[active_envelope].SetAttackLength(block->cv_slider[1] + block->cv[1]);
+    eg[active_envelope].SetAttackCurve (block->pot[1]);
+  }
+  if (slider_enabled[2]) {
+    eg[active_envelope].SetHoldLength(block->cv_slider[2] + block->cv[2]);
+  }
+  if (slider_enabled[3]) {
+    eg[active_envelope].SetDecayLength(block->cv_slider[3] + block->cv[3]);
+    eg[active_envelope].SetDecayCurve  (block->pot[3]);
+  }
+  if (slider_enabled[4]) {
+    eg[active_envelope].SetSustainLevel(block->cv_slider[4] + block->cv[4]);
+  }
+  if (slider_enabled[5]) {
+    eg[active_envelope].SetReleaseLength(block->cv_slider[5] + block->cv[5]);
+    eg[active_envelope].SetReleaseCurve(block->pot[5]);
+  }
```

Comment:
Oh sorry should have caught this earlier. `block->cv_slider` is already the sum of `block->cv` and `block->slider` (constrained to [-1, 2)). So I think you just want `block->cv_slider` here. Alternatively, if you end up wanting use CV for something else, just `block->slider`.